clusterloader2/api/validation/validation.go

```diff
@@ -0,0 +1,180 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/perf-tests/clusterloader2/api"
+
+	_ "k8s.io/kubernetes/pkg/apis/core/install"
+)
+
+// VerifyConfig does a simple validation for config and all included structures.
+// Returns list of all not satisfied conditions.
+func VerifyConfig(c *api.Config) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.AutomanagedNamespaces < 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("automanagedNamespaces"), c.AutomanagedNamespaces, "must be non-negative"))
+	}
+	for i := range c.TuningSets {
+		allErrs = append(allErrs, verifyTuningSet(&c.TuningSets[i], field.NewPath("tuningSets").Index(i))...)
+	}
+	for i := range c.Steps {
+		allErrs = append(allErrs, verifyStep(&c.Steps[i], field.NewPath("steps").Index(i))...)
+	}
+	return allErrs
+}
+
+func verifyStep(s *api.Step, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i := range s.Phases {
+		allErrs = append(allErrs, verifyPhase(&s.Phases[i], fldPath.Child("phases").Index(i))...)
+	}
+	for i := range s.Measurements {
+		allErrs = append(allErrs, verifyMeasurement(&s.Measurements[i], fldPath.Child("measurements").Index(i))...)
+	}
+	isPhasesEmpty := len(s.Phases) == 0
+	isMeasurementsEmpty := len(s.Measurements) == 0
+	if isPhasesEmpty && isMeasurementsEmpty {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("measurements"), s.Measurements, "can't be empty with empty phases"))
+	}
+	if !isPhasesEmpty && !isMeasurementsEmpty {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("measurements"), s.Measurements, "can't be non-empty with non-empty phases"))
+	}
+	return allErrs
+}
+
+func verifyPhase(p *api.Phase, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if p.NamespaceRange != nil {
+		allErrs = append(allErrs, verifyNamespaceRange(p.NamespaceRange, fldPath.Child("namespaceRange"))...)
+	}
+	if p.ReplicasPerNamespace < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicasPerNamespace"), p.ReplicasPerNamespace, "must be non-negative"))
+	}
+	for i := range p.ObjectBundle {
+		allErrs = append(allErrs, verifyObject(&p.ObjectBundle[i], fldPath.Child("objectBundle").Index(i))...)
+	}
+	return allErrs
+}
+
+func verifyObject(o *api.Object, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if o.ObjectType.Kind == "Service" {
```

Comment:
Please don't do anything like that - such validation will be done by underlying k8s cluster.
The goal of cluster-loader project is not to copy validation from k8s.

All of this should be removed.

Reply:
I'm changing it to IsDNS1123Subdomain verification. This one should verify names correctly.